python/ml4ir/applications/ranking/tests/test_feature_layer.py

```diff
@@ -282,27 +299,114 @@ def test_categorical_embedding_with_vocabulary_file(self):
         assert not tf.reduce_all(tf.equal(categorical_embedding[0], categorical_embedding[1]))
         assert tf.reduce_all(tf.equal(categorical_embedding[4], categorical_embedding[5]))
 
-        # Strings group_0 and group_2 should NOT result in the same embedding because they use a default one-to-one vocabulary mapping
+        # Strings domain_0 and domain_2 should NOT result in the same embedding because they use a default one-to-one vocabulary mapping
         assert not tf.reduce_all(tf.equal(categorical_embedding[0], categorical_embedding[3]))
         assert not tf.reduce_all(tf.equal(categorical_embedding[3], categorical_embedding[4]))
 
-    def test_categorical_indicator_with_vocabulary_file(self):
+    def test_categorical_embedding_with_vocabulary_file_with_ids_and_dropout(self):
```

Comment:
@mohazahran This is a unittest that captures what is happening in the function. The goal is to learn categorical embeddings, but "drop" the categorical tokens at a `dropout_rate` to the OOV token so that the OOV embedding is learned across all categorical values.